src/Endpoints/class-analytics-post-detail-api-proxy.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Endpoints: Parse.ly `/analytics/post/detail` API proxy endpoint class
+ *
+ * @package Parsely
+ * @since   3.6.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Endpoints;
+
+use stdClass;
+use WP_REST_Request;
+
+/**
+ * Configures the `/analytics/post/detail` REST API endpoint.
+ */
+final class Analytics_Post_Detail_API_Proxy extends Base_API_Proxy {
+
+	/**
+	 * Registers the endpoint's WP REST route.
+	 */
+	public function run(): void {
+		$this->register_endpoint( '/analytics/post/detail' );
+	}
+
+	/**
+	 * Cached "proxy" to the Parse.ly `/analytics/post/detail` API endpoint.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 */
+	public function get_items( WP_REST_Request $request ): stdClass {
+		return $this->get_data( $request );
+	}
+
+	/**
+	 * Generates the final data from the passed response.
+	 *
+	 * @param array<string, mixed> $response The response received by the proxy.
+	 * @return array<stdClass> The generated data.
+	 */
+	protected function generate_data( array $response ): array {
+		$stats_base_url = trailingslashit( 'https://dash.parsely.com/' . $this->parsely->get_api_key() ) . 'find';
+
+		$result = array_map(
+			static function( stdClass $item ) use ( $stats_base_url ) {
+				return (object) array(
+					'avgEngaged' => self::get_duration( (float) $item->avg_engaged ),
+					'statsUrl'   => $stats_base_url . '?url=' . rawurlencode( $item->url ),
+					'url'        => $item->url,
+					'views'      => number_format_i18n( $item->metrics->views ),
+					'visitors'   => number_format_i18n( $item->metrics->visitors ),
+				);
+			},
+			$response
+		);
+
+		return $result;
+	}
+
+	/**
+	 * Returns the passed number as a time duration.
+	 *
+	 * @since 3.6.0
+	 *
+	 * @param float $time The time as a float number.
+	 * @return string The resulting formatted time duration.
+	 */
+	private function get_duration( float $time ) {
+		$decimal_value = round( fmod( $time, 1 ) * 60 );
+		return sprintf( '%2d:%02d', (int) $time, $decimal_value );
+	}
```

Comment:
I think this could benefit from unit tests - for instance, a `$time` of `999.999` returns a string of `999:60`, which doesn't make a lot of sense. If you're going to format it into `mm:ss` then it needs some more guards.

Also, updating the docs to indicate an example of the incoming format, and what the returned format should look like (and that it is meant to be `mm:ss`) would be useful for understanding the intent of the code.

Reply:
I've improved the function and its docs. I did not write unit tests for it, but marking this as resolved.